plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/BaseIcebergConnectorTest.java

```diff
@@ -1995,6 +2000,72 @@ public void testIncorrectIcebergFileSizes()
         dropTable("test_iceberg_file_size");
     }
 
+    @Test
+    public void testSplitPruningForFilterOnPartitionColumn()
+    {
+        String tableName = "nation_partitioned_pruning";
+
+        // disable writes redistribution to have predicatable number of files written per partition (one).
+        Session noRedistributeWrites = Session.builder(getSession())
+                .setSystemProperty("redistribute_writes", "false")
+                .build();
+
+        assertUpdate(noRedistributeWrites, "CREATE TABLE " + tableName + " WITH (partitioning = ARRAY['regionkey']) AS SELECT * FROM nation", 25);
+
+        ResultWithQueryId<MaterializedResult> selectAllPartitionsResult = getDistributedQueryRunner().executeWithQueryId(
+                getSession(),
+                "SELECT * FROM " + tableName);
+
+        ResultWithQueryId<MaterializedResult> selectSinglePartitionResult = getDistributedQueryRunner().executeWithQueryId(
+                getSession(),
+                "SELECT * FROM " + tableName + " WHERE regionkey = 3");
+
+        ResultWithQueryId<MaterializedResult> selectTwoPartitionsResult = getDistributedQueryRunner().executeWithQueryId(getSession(),
+                "SELECT * FROM " + tableName + " WHERE regionkey < 2");
+
+        verifySplitCount(selectAllPartitionsResult.getQueryId(), 5);
+        verifySplitCount(selectSinglePartitionResult.getQueryId(), 1);
+        verifySplitCount(selectTwoPartitionsResult.getQueryId(), 2);
+
+        dropTable(tableName);
```

Comment:
that's exuberant. we don't need to test that drop table drops a table
`assertQuery("DROP TABLE " + tableName)` would suffice